src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/Cas2ApplicationsController.kt

```diff
@@ -0,0 +1,157 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.ApplicationsTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.getFullInfoForPersonOrThrow
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.getInfoForPersonOrThrowInternalServerError
+import java.net.URI
+import java.util.UUID
+import javax.transaction.Transactional
+
+@Service
+class Cas2ApplicationsController(
+  private val httpAuthService: HttpAuthService,
+  private val applicationService: ApplicationService,
+  private val applicationsTransformer: ApplicationsTransformer,
+  private val objectMapper: ObjectMapper,
+  private val offenderService: OffenderService,
+  private val userService: UserService,
+) : Cas2ApiDelegate {
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  override fun cas2ApplicationsGet(): ResponseEntity<List<ApplicationSummary>> {
+    val user = userService.getUserForRequest()
+
+    val applications = applicationService.getAllApplicationsForUsername(
+      user
+        .deliusUsername,
+      ServiceName.cas2,
+    )
+
+    return ResponseEntity.ok(applications.map { getPersonDetailAndTransformToSummary(it, user) })
+  }
+
+  override fun cas2ApplicationsApplicationIdGet(applicationId: UUID):
+    ResponseEntity<Application> {
+    val user = userService.getUserForRequest()
+
+    val application = when (val applicationResult = applicationService.getApplicationForUsername(applicationId, user.deliusUsername)) {
+      is AuthorisableActionResult.NotFound -> null
+      is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
+      is AuthorisableActionResult.Success -> applicationResult.entity
+    }
+
+    if (application != null) {
+      return ResponseEntity.ok(getPersonDetailAndTransform(application, user))
+    }
+    throw NotFoundProblem(applicationId, "Application")
+  }
+
+  @Transactional
+  override fun cas2ApplicationsPost(body: NewApplication):
+    ResponseEntity<Application> {
+    val deliusPrincipal = httpAuthService.getDeliusPrincipalOrThrow()
+    val user = userService.getUserForRequest()
+
+    val personInfo = offenderService.getFullInfoForPersonOrThrow(body.crn, user)
+
+    val applicationResult = applicationService.createCas2Application(body.crn, user, deliusPrincipal.token.tokenValue)
+
+    val application = when (applicationResult) {
+      is ValidatableActionResult.GeneralValidationError -> throw BadRequestProblem(errorDetail = applicationResult.message)
+      is ValidatableActionResult.FieldValidationError -> throw BadRequestProblem(invalidParams = applicationResult.validationMessages)
+      is ValidatableActionResult.ConflictError -> throw ConflictProblem(id = applicationResult.conflictingEntityId, conflictReason = applicationResult.message)
+      is ValidatableActionResult.Success -> applicationResult.entity
+    }
+
+    return ResponseEntity
+      .created(URI.create("/applications/${application.id}"))
```

Comment:
👍 thanks! I failed to understand this properly -- it's setting the `location` header in the response so that the UI knows where to find the newly created resource:

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html#created(java.net.URI)

<img width="429" alt="created" src="https://github.com/ministryofjustice/hmpps-approved-premises-api/assets/20245/0b568c97-a671-4288-aa8c-d24339ee79df">